docs/changelog.md

```diff
@@ -1,5 +1,42 @@
 # Changelog
 
+## 1.12.0 (2025-05-13)
+
+
+### Major changes
+
+- The build system has been updated to use `pyproject.toml` instead of `setup.py`
+- Support for Python 3.9 has been dropped. The minimum supported Python version is now
+  3.10; the minimum supported NumPy version is 1.22; the minimum supported `ndonnx`
+  version is 0.10.1.
+- The `linalg` extension works correctly with `pytorch==2.7`.
+- Multiple improvements to handling of `device` arguments in `numpy`, `cupy`, `torch`,
+  and `dask` backends. Support for multiple devices is still relatively immature,
+  and rough edges can be expected. Please report any issues you encounter.
```

Comment:
```suggestion
- Multiple improvements to handling of devices in CuPy and PyTorch backends.
  Support for multiple devices in CuPy is still immature and you should use
  context managers rather than relying on input-output device propagation or
  on the `device` parameter.
```
We didn't do anything in numpy or dask - they work just as before.
In cupy, we didn't do anything about the device argument in the end (#293 is DNM). We did tweak `to_device` though.
To the best of my knowledge pytorch now works 100%.